src/frontend/screens/MapScreen/GPSPermissions/GPSPermissionsModal.tsx

```diff
@@ -1,34 +1,86 @@
 import {GPSPermissionsEnabled} from './GPSPermissionsEnabled';
 import * as Location from 'expo-location';
 import {useGPSModalContext} from '../../../contexts/GPSModalContext';
 import {useTabNavigationStore} from '../../../hooks/useTabNavigationStore';
 import {BottomSheetModal, BottomSheetView} from '@gorhom/bottom-sheet';
 import {TAB_BAR_HEIGHT} from '../../../Navigation/Stack/AppScreens';
-import {StyleSheet} from 'react-native';
-import {useFocusEffect} from '@react-navigation/native';
+import {StyleSheet, Linking, View} from 'react-native';
+import {GPSBackgroundPermissionDisabled} from './GPSBackgroundPermissionDisabled';
+import {Loading} from '../../../sharedComponents/Loading';
+
+const handleOpenSettings = () => {
+  Linking.sendIntent('android.settings.LOCATION_SOURCE_SETTINGS');
+};
 
 export const GPSPermissionsModal = React.memo(() => {
   const {setCurrentTab} = useTabNavigationStore();
-  const [backgroundStatus] = Location.useBackgroundPermissions();
-  const [foregroundStatus] = Location.useForegroundPermissions();
 
-  const [isGranted, setIsGranted] = useState<boolean | null>(null);
+  const [foregroundPermission, setForegroundPermission] =
+    React.useState<Location.LocationPermissionResponse | null>(null);
+  const [backgroundPermission, setBackgroundPermission] =
+    React.useState<Location.LocationPermissionResponse | null>(null);
   const {bottomSheetRef} = useGPSModalContext();
 
-  useEffect(() => {
-    if (backgroundStatus && foregroundStatus && isGranted === null) {
-      setIsGranted(backgroundStatus.granted && foregroundStatus.granted);
-    }
-  }, [backgroundStatus, foregroundStatus, isGranted]);
+  console.log({foregroundPermission: foregroundPermission?.granted});
+
+  React.useEffect(() => {
+    Location.getForegroundPermissionsAsync().then(permission =>
+      setForegroundPermission(permission),
+    );
+
+    Location.getBackgroundPermissionsAsync().then(permission =>
+      setBackgroundPermission(permission),
+    );
+  }, []);
 
   const onBottomSheetDismiss = () => {
     setCurrentTab('Map');
   };
-  useFocusEffect(() => {
-    return () => bottomSheetRef?.current?.close();
-  });
+
+  async function askForegroundLocationPermission() {
+    if (foregroundPermission!.canAskAgain) {
+      const permission = await Location.requestForegroundPermissionsAsync();
+      setForegroundPermission(permission);
+    } else {
+      handleOpenSettings();
+    }
+  }
+
+  async function askBackgroundLocationPermission() {
+    if (backgroundPermission!.canAskAgain) {
+      const permission = await Location.requestBackgroundPermissionsAsync();
+      setBackgroundPermission(permission);
+    } else {
+      handleOpenSettings();
+    }
+  }
+
+  const renderContent = () => {
+    if (!foregroundPermission || !backgroundPermission) {
+      return (
+        <View style={{display: 'flex', minHeight: 200}}>
+          <Loading />
+        </View>
```

Comment:
there is a small (but unlikely) chance that the user can open the track modal BEFORE the permission have been loaded. I needed to set a height to view in order for this to now just be a sliver of a bottom sheet. So I arbitrarly set 200, but I am open to suggestions if we don't want to use an arbitrary number

Reply:
have unfortunately run into this before 🥲